File: Longest Substring Without Repeating Characters.py
class Solution:
	# @return an integer
	def lengthOfLongestSubstring(self, s):
		longest = 0
		start = 0
		end = 0
		test = {}
		while (end < len(s)):
			if (not s[end] in test):
				test[s[end]] = True
			else:
				while (s[start] != s[end]):
					del test[s[start]]
					start = start + 1
				start = start + 1
			end = end + 1
			if (end - start > longest):
				longest = end - start
		return longest

File: Longest Substring Without Repeating Characters.java
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        boolean[] test = new boolean [256];
        int longest = 0;
        int start = 0;  // include
        int end = 0;    // exclude
        while (end < s.length()) {
            if (!test[s.charAt(end)]) {
                test[s.charAt(end)] = true;
            }
            else {
                while (s.charAt(start) != s.charAt(end)) {
                    test[s.charAt(start)] = false;
                    start++;
                }
                start++;
            }
            end++;
            if (end - start > longest)
                longest = end - start;
        }
        return longest;
    }
}